db/trie/context.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2018 IoTeX
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package trie
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/pkg/errors"
+
+	"github.com/iotexproject/iotex-core/db"
+	"github.com/iotexproject/iotex-core/pkg/hash"
+	iproto "github.com/iotexproject/iotex-core/proto"
+)
+
+type sameKeyLenTrieContextKey struct{}
+
+// SameKeyLenTrieContext provides the trie operation with auxiliary information.
+type SameKeyLenTrieContext struct {
```

Comment:
it is good trick to group these into a context
i think we can simply name it TrieContext though? b/c the context include common vars like Bucket, DB, CB, KeyLength. 
KeyLength is an important var, we can use it to enforce all keys passed to trie has same length, on the other hand it is one of context. we shouldn't use it to name the entire context

Reply:
I am going to delete this file in follow up PR